server/app/views/tags/IconElementTagProcessor.java

```diff
@@ -0,0 +1,44 @@
+package views.tags;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import controllers.AssetsFinder;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import org.thymeleaf.context.ITemplateContext;
+import org.thymeleaf.model.IProcessableElementTag;
+import org.thymeleaf.processor.element.AbstractElementTagProcessor;
+import org.thymeleaf.processor.element.IElementTagStructureHandler;
+import org.thymeleaf.templatemode.TemplateMode;
+
+public final class IconElementTagProcessor extends AbstractElementTagProcessor {
+  private final AssetsFinder assetsFinder;
+
+  public IconElementTagProcessor(String dialectPrefix, AssetsFinder assetsFinder) {
+    super(TemplateMode.HTML, dialectPrefix, "icon", true, null, false, 10);
+    this.assetsFinder = checkNotNull(assetsFinder);
+  }
+
+  @Override
+  protected void doProcess(
+      ITemplateContext context,
+      IProcessableElementTag tag,
+      IElementTagStructureHandler structureHandler) {
+    String html =
+        """
+            <img class="%s" src="%s" role="img" alt="" aria-hidden="true" />
+        """;
```

Comment:
This template should be converted to j2html.

While we're moving away from the UI being written with it I don't see a need to remove it completely in target situations. Long term if we get to a point where this is the only thing that uses it we can rewrite it then.